include/libtorrent/kademlia/node.hpp

```diff
@@ -285,7 +285,7 @@ class TORRENT_EXTRA_EXPORT node : boost::noncopyable
 	udp_socket_interface* m_sock;
 	counters& m_counters;
 
-	boost::scoped_ptr<dht_storage_interface> m_storage;
+	dht_storage_interface* m_storage;
```

Comment:
a reference also makes it obvious that there's no owning relationship. ideally a raw pointer would also never mean owning, but I imagine a lot of c++ programmers have a reflex of worrying about raw pointers and ownership